src/finiteVolume/processes/thermophoreticDiffusion.cpp

```diff
@@ -0,0 +1,97 @@
+    flow.RegisterRHSFunction(ThermophoreticDiffusionVariableFlux,
+                             &viscosityTemperatureFunction,
+                             CompressibleFlowFields::DENSITY_YI_FIELD,
+                             {CompressibleFlowFields::EULER_FIELD, CompressibleFlowFields::DENSITY_YI_FIELD},
+                             {CompressibleFlowFields::TEMPERATURE_FIELD});
+    flow.RegisterRHSFunction(ThermophoreticDiffusionVariableFlux,
+                             &viscosityTemperatureFunction,
+                             CompressibleFlowFields::DENSITY_PROGRESS_FIELD,
+                             {CompressibleFlowFields::EULER_FIELD, CompressibleFlowFields::DENSITY_PROGRESS_FIELD},
+                             {CompressibleFlowFields::TEMPERATURE_FIELD});
+
+    viscosityTemperatureFunction = transportModel->GetTransportTemperatureFunction(eos::transport::TransportProperty::Viscosity, flow.GetSubDomain().GetFields());
+}
+
+PetscErrorCode ablate::finiteVolume::processes::ThermophoreticDiffusion::ThermophoreticDiffusionEnergyFlux(PetscInt dim, const PetscFVFaceGeom *fg, const PetscInt uOff[], const PetscInt uOff_x[],
+                                                                                                           const PetscScalar field[], const PetscScalar grad[], const PetscInt aOff[],
+                                                                                                           const PetscInt aOff_x[], const PetscScalar aux[], const PetscScalar gradAux[],
+                                                                                                           PetscScalar flux[], void *ctx) {
+    PetscFunctionBeginUser;
+    // this order is based upon the order that they are passed into RegisterRHSFunction
+    const int EULER_FIELD = 0;
+    const int DENSITY_YI_FIELD = 1;
+    const int TEMPERATURE_FIELD = 0;
+
+    auto viscosityFunction = (eos::ThermodynamicTemperatureFunction *)ctx;
+
+    // get the current density from euler
+    const PetscReal density = field[uOff[EULER_FIELD] + CompressibleFlowFields::RHO];
+
+    // compute the viscosity on the boundary
+    PetscReal temperature = aux[aOff[TEMPERATURE_FIELD]];
+    PetscReal mu;
+    PetscCall(viscosityFunction->function(field, temperature, &mu, viscosityFunction->context.get()));
+
+    // compute the coefficients for carbon and ndd (note that the negatives cancel).  Note that carbon is always assumed to be at the zero index
+    PetscReal coefficient = eos::TChemSoot::ComputeSolidCarbonSensibleEnthalpy(temperature) * field[uOff[DENSITY_YI_FIELD]] * 0.5 * mu / (density * temperature); /*hc*density*Yi*0.5*mu/(rho*T)*/
+
+    // The current energy flux should be zero at the start
+    for (PetscInt d = 0; d < dim; ++d) {
+        // flux( (carbonCoefficient +  nddCoefficient) * (- dT/dx - dT/dy - dT/dz) . n A
+        flux[CompressibleFlowFields::RHOE] += -fg->normal[d] * gradAux[aOff_x[TEMPERATURE_FIELD] + d];
```

Comment:
I'm not exactly sure the notation here, but I'm assuming that this is the normal component of the face in the wanted direction * the gradient of Temperature out of the face * the area. Should there be a density multiplying this along with the coefficient since the convective flux is (rho*coeff_{Thermo}*Grad(T)) over the cell? face?